pkg/segment/query/processor/statscommand_test.go

```diff
@@ -379,3 +379,40 @@ func Test_ProcessSegmentStats(t *testing.T) {
 	assert.Equal(t, 1, len(actualAvgRes))
 	assert.Equal(t, expectedAvgRes, actualAvgRes)
 }
+
+func Test_ProcessGroupByRequestFullBuffer(t *testing.T) {
+	config.InitializeTestingConfig(t.TempDir())
+	config.GetRunningConfig().UseNewPipelineConverted = true
+
+	knownValues := getTestData()
+	processor := getGroupByProcessor()
+	assert.Equal(t, 0, len(processor.bucketKeyWorkingBuf))
+
+	col1Values := knownValues["col1"]
+	factorSize := 5
+
+	for i := 0; i < len(col1Values); i++ {
+		byteVal := make([]byte, utils.MAX_RECORD_SIZE*3)
```

Comment:
Where is this 3 coming from? Is it based off the `factorSize`? Using `factorSize-1` could make that more clear

Also, is this `factorSize` arbitrary, or is it based on some other property of the test?

Reply:
oh my bad, that 3 was supposed to be `factorSize` when I was testing I created the variable and set it to 5 but forgot to replace it here.

Reply:
yes, factorSize is arbitary, but basically any value greater than the len(groupbyColumns). i will add that as a comment